### PR TITLE
Adding comment from source code to DetectionOutputLayer class documentation

### LINKED ISSUE #14645: DetectionOutput layer doc
FasterRCnn use DetectionOutput as inference results. In source code it is written

[        // num() and channels() are 1.
        // Since the number of bboxes to be kept is unknown before nms, we manually
        // set it to maximal number of detections, [keep_top_k] parameter multiplied by batch size.
        // Each row is a 7 dimension std::vector, which stores
// [image_id, label, confidence, xmin, ymin, xmax, ymax]](https://github.com/LaurentBerger/opencv/blob/6c6dd16acf47da980f1f33ce578b02868f112b47/modules/dnn/src/layers/detection_output_layer.cpp#L221-L222)

Can we add this comment in [detectionOutput layer doc](https://docs.opencv.org/trunk/da/dbd/classcv_1_1dnn_1_1DetectionOutputLayer.html)?

### PR DESCRIPTION
**Pull request by SoheibKadi**
Adds a comment written in the source code for bool getMemoryShapes into the DetectionOutputLayer class documentation, as requested in this issue:

resolves #14645

```
force_builders_only=Docs
```

### CODE REVIEW

**Comment by dkurt** [comment]
please do not duplicate issue but help to resolve it.

File: modules/dnn/include/opencv2/dnn/all_layers.hpp [hunk 1]
```diff
@@ -600,6 +600,15 @@ CV__DNN_INLINE_NS_BEGIN
         static Ptr<RegionLayer> create(const LayerParams& params);
     };
 
+    /**
+     * @brief \f$ L_p \f$ - detection output layer.
```
Comment (dkurt) [inline modules/dnn/include/opencv2/dnn/all_layers.hpp:604]: What is Lp?

File: modules/dnn/include/opencv2/dnn/all_layers.hpp [hunk 2]
```diff
@@ -600,6 +600,15 @@ CV__DNN_INLINE_NS_BEGIN
...
+     *
+     * num() and channels() are 1.
```
Comment (dkurt) [inline modules/dnn/include/opencv2/dnn/all_layers.hpp:606]: What is num and channels? Just specify how output shape is computed

File: modules/dnn/include/opencv2/dnn/all_layers.hpp (hunk 1 above)
Comment (SoheibKadi) [reply modules/dnn/include/opencv2/dnn/all_layers.hpp:604]: Oh. I copied the comment from below and left that part. Removing it now.

File: modules/dnn/include/opencv2/dnn/all_layers.hpp (hunk 2 above)
Comment (SoheibKadi) [reply modules/dnn/include/opencv2/dnn/all_layers.hpp:606]: OK

Comment (SoheibKadi) [reply modules/dnn/include/opencv2/dnn/all_layers.hpp:606]: Is it better like this:

/**
     * @brief detection output layer.
     *
     * The layer size is: @f$ (1 \times 1 \times N \times 7) @f$
     *    where N is the number of detections after nms, and each row is:
     *    [image_id, label, confidence, xmin, ymin, xmax, ymax]
     */

Comment (dkurt) [reply modules/dnn/include/opencv2/dnn/all_layers.hpp:606]: number of detections multiplied by batch size?

Comment (SoheibKadi) [reply modules/dnn/include/opencv2/dnn/all_layers.hpp:606]: Yes. At least that's what I see in code.

**Comment by alalek** [comment]
This patch should go into [3.4 branch](https://github.com/opencv/opencv/wiki/Branches) first.
We will merge changes from 3.4 into master regularly (weekly/bi-weekly).

So, please:
- change "base" branch of this PR: master => 3.4 (use "Edit" button near PR title)
- rebase your commits from master onto 3.4 branch. For example:
    `git rebase -i --onto upstream/3.4 upstream/master`
    (check list of your commits, save and quit (Esc + "wq" + Enter)
    where `upstream` is configured by following [this GitHub guide](https://help.github.com/articles/configuring-a-remote-for-a-fork/) and fetched (`git fetch upstream`).
- push rebased commits into source branch of your fork (with `--force` option)

Note: no needs to re-open PR, apply changes "inplace".

File: modules/dnn/include/opencv2/dnn/all_layers.hpp [hunk 3]
```diff
@@ -598,6 +598,13 @@ CV__DNN_EXPERIMENTAL_NS_BEGIN
         static Ptr<RegionLayer> create(const LayerParams& params);
     };
 
+    /**
+     * @brief detection output layer.
+     *
+     * The layer size is: @f$ (1 \times 1 \times N \times 7) @f$
+     *    where N is the number of detections after nms, and each row is:
+     *    [image_id, label, confidence, xmin, ymin, xmax, ymax]
```
Comment (dkurt) [inline modules/dnn/include/opencv2/dnn/all_layers.hpp:606]: Please clarify that `image id in batch`

**Comment by asmorkalov** [comment]
@SoheibKadi friendly reminder.

**Pull request merged**